### ISSUE TITLE
problems with https://google.com or nordvpn servers were hacked with MITM?

### ISSUE THREAD

**Issue by tm4ig** [NordSecurity/nordvpn-linux#551]
Today I have sometimes problems with https://google.com when nordvpn is connected.
https://google.com sometimes does not opening 
In the curl I have sometimes this error when nordvpn is working
```
curl https://google.com -v
*   Trying 192.0.0.88:443...
* Connected to google.com (192.0.0.88) port 443 (#0)
* ALPN, offering h2
* ALPN, offering http/1.1
*  CAfile: /etc/ssl/certs/ca-certificates.crt
*  CApath: /etc/ssl/certs
* TLSv1.0 (OUT), TLS header, Certificate Status (22):
* TLSv1.3 (OUT), TLS handshake, Client hello (1):
* TLSv1.0 (OUT), TLS header, Unknown (21):
* TLSv1.3 (OUT), TLS alert, decode error (562):
* error:0A000126:SSL routines::unexpected eof while reading
* Closing connection 0
curl: (35) error:0A000126:SSL routines::unexpected eof while reading
```
Also google.com is resolving to strange ip address 192.0.0.88 with nordvpn dns servers 

```
resolvectl status nordlynx
Link 4 (nordlynx)
    Current Scopes: DNS
         Protocols: +DefaultRoute +LLMNR -mDNS -DNSOverTLS DNSSEC=allow-downgrade/supported
Current DNS Server: 103.86.96.100
       DNS Servers: 103.86.96.100 103.86.99.100
        DNS Domain: ~.


host google.com 103.86.96.100
Using domain server:
Name: 103.86.96.100
Address: 103.86.96.100#53
Aliases: 

google.com has address 192.0.0.88
google.com mail is handled by 10 smtp.google.com.

```

whois says that 192.0.0.88 is not google ip 

Without nordvpn i have not any problems with google.com and google.com was resolved to google ip 172.217.168.14

my nordvpn config
```
nordvpn status
Status: Connected
Server: Germany #1047
Hostname: de1047.nordvpn.com
IP: 91.214.65.121
Country: Germany
City: Frankfurt
Current technology: NORDLYNX
Current protocol: UDP
Transfer: 2.81 MiB received, 372.93 KiB sent
Uptime: 1 minute 56 seconds

nordvpn settings
Technology: NORDLYNX
Firewall: disabled
Firewall Mark: 0xe1f1
Routing: enabled
Analytics: disabled
Kill Switch: disabled
Threat Protection Lite: disabled
Notify: disabled
Tray: disabled
Auto-connect: disabled
IPv6: disabled
Meshnet: disabled
DNS: disabled
LAN Discovery: disabled
Virtual Location: enabled
Allowlisted ports:
Allowlisted subnets:

nordvpn version
NordVPN Version 3.18.3

 cat /etc/os-release 
PRETTY_NAME="Ubuntu 22.04.4 LTS"
NAME="Ubuntu"
VERSION_ID="22.04"
VERSION="22.04.4 LTS (Jammy Jellyfish)"
VERSION_CODENAME=jammy
ID=ubuntu
ID_LIKE=debian
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"



```

**Comment by tm4ig** [comment]
Why does nordpn dns say that google.com has ip 192.0.0.88 
https://forum.manjaro.org/t/nordvpn-bin-and-steamclient/151160/30?page=3
https://www.reddit.com/r/nordvpn/comments/13ii3zk/nordvpn_dns_servers_resolve_googlecom_into_a/
and why does google.com say ssl errors with nordvpn?

**Comment by tm4ig** [comment]
with custom dns for active nordvpn (for example 1.1.1.1) I have not any problems for google.com 
so problem is in nordvpn fake google.com ip 192.0.0.88 and nordvpn dns servers

**Comment by tm4ig** [comment]
With 192.0.0.88 as target ip for google.com address  works intermittently
```
curl --resolve google.com:443:192.0.0.88 https://google.com
<HTML><HEAD><meta http-equiv="content-type" content="text/html;charset=utf-8">
<TITLE>301 Moved</TITLE></HEAD><BODY>
<H1>301 Moved</H1>
The document has moved
<A HREF="https://www.google.com/">here</A>.
</BODY></HTML>

 curl --resolve google.com:443:192.0.0.88 https://google.com
curl: (35) error:0A000126:SSL routines::unexpected eof while reading
```


```
 192.0.0.88 as target host doesn't work at all
curl https://192.0.0.88 -k
curl: (35) error:0A000126:SSL routines::unexpected eof while reading

curl https://192.0.0.88 -k
curl: (35) error:0A000126:SSL routines::unexpected eof while reading

```

**Comment by tm4ig** [comment]
in the firefox I have  PR_END_OF_FILE_ERROR error for google.com with nordvpn

**Comment by mariusSincovici** [comment]
Hi,
Thanks for your feedback.
The fact that google.com resolves to 192.0.0.88, it is a feature and there will be an explanation page for this.

Regarding the certificate issue, unfortunately I cannot reproduce this and tried on Ubuntu 22 and 24.
I did saw a difference between my output and yours:
```
* TLSv1.3 (OUT), TLS handshake, Client hello (1):
* TLSv1.2 (IN), TLS header, Certificate Status (22): <---- you get: * TLSv1.0 (OUT), TLS header, Unknown (21):
* TLSv1.3 (IN), TLS handshake, Server hello (2):
```
Could you check the output of `curl` when not using the VPN, mainly the line after `TLSv1.3 (OUT), TLS handshake, Client hello (1):`?
Also while connected to the VPN please check if `openssl s_client -connect google.com:443 -tls1_2` works fine?
Thanks

**Comment by tm4ig** [comment]
@mariusSincovici 

nordvpn ON, google.com curl OK answer
```
curl -v https://google.com -s > /dev/null 
*   Trying 192.0.0.88:443...
* Connected to google.com (192.0.0.88) port 443 (#0)
* ALPN, offering h2
* ALPN, offering http/1.1
*  CAfile: /etc/ssl/certs/ca-certificates.crt
*  CApath: /etc/ssl/certs
* TLSv1.0 (OUT), TLS header, Certificate Status (22):
} [5 bytes data]
* TLSv1.3 (OUT), TLS handshake, Client hello (1):
} [512 bytes data]
* TLSv1.2 (IN), TLS header, Certificate Status (22):
{ [5 bytes data]
* TLSv1.3 (IN), TLS handshake, Server hello (2):
{ [122 bytes data]
* TLSv1.2 (IN), TLS header, Finished (20):
{ [5 bytes data]
* TLSv1.2 (IN), TLS header, Supplemental data (23):
{ [5 bytes data]
* TLSv1.3 (IN), TLS handshake, Encrypted Extensions (8):
{ [15 bytes data]
* TLSv1.3 (IN), TLS handshake, Certificate (11):
{ [6300 bytes data]
* TLSv1.3 (IN), TLS handshake, CERT verify (15):
{ [79 bytes data]
* TLSv1.3 (IN), TLS handshake, Finished (20):
{ [52 bytes data]
* TLSv1.2 (OUT), TLS header, Finished (20):
} [5 bytes data]
* TLSv1.3 (OUT), TLS change cipher, Change cipher spec (1):
} [1 bytes data]
* TLSv1.2 (OUT), TLS header, Supplemental data (23):
} [5 bytes data]
* TLSv1.3 (OUT), TLS handshake, Finished (20):
} [52 bytes data]
* SSL connection using TLSv1.3 / TLS_AES_256_GCM_SHA384
* ALPN, server accepted to use h2
* Server certificate:
*  subject: CN=*.google.com
*  start date: Jul 30 12:32:53 2024 GMT
*  expire date: Oct 22 12:32:52 2024 GMT
*  subjectAltName: host "google.com" matched cert's "google.com"
*  issuer: C=US; O=Google Trust Services; CN=WR2
*  SSL certificate verify ok.
* Using HTTP2, server supports multiplexing
* Connection state changed (HTTP/2 confirmed)
* Copying HTTP/2 data in stream buffer to connection buffer after upgrade: len=0
* TLSv1.2 (OUT), TLS header, Supplemental data (23):
} [5 bytes data]
* TLSv1.2 (OUT), TLS header, Supplemental data (23):
} [5 bytes data]
* TLSv1.2 (OUT), TLS header, Supplemental data (23):
} [5 bytes data]
* Using Stream ID: 1 (easy handle 0x55934e8cceb0)
* TLSv1.2 (OUT), TLS header, Supplemental data (23):
} [5 bytes data]
> GET / HTTP/2
> Host: google.com
> user-agent: curl/7.81.0
> accept: */*
> 
* TLSv1.2 (IN), TLS header, Supplemental data (23):
{ [5 bytes data]
* TLSv1.3 (IN), TLS handshake, Newsession Ticket (4):
{ [282 bytes data]
* TLSv1.3 (IN), TLS handshake, Newsession Ticket (4):
{ [282 bytes data]
* old SSL session ID is stale, removing
* TLSv1.2 (IN), TLS header, Supplemental data (23):
{ [5 bytes data]
* TLSv1.2 (OUT), TLS header, Supplemental data (23):
} [5 bytes data]
* TLSv1.2 (IN), TLS header, Supplemental data (23):
{ [5 bytes data]
* TLSv1.2 (IN), TLS header, Supplemental data (23):
{ [5 bytes data]
< HTTP/2 301 
< location: https://www.google.com/
< content-type: text/html; charset=UTF-8
< content-security-policy-report-only: object-src 'none';base-uri 'self';script-src 'nonce-wtMpnqmoLZKTVYPL0Jxt8w' 'strict-dynamic' 'report-sample' 'unsafe-eval' 'unsafe-inline' https: http:;report-uri https://csp.withgoogle.com/csp/gws/other-hp
< date: Mon, 12 Aug 2024 09:51:46 GMT
< expires: Wed, 11 Sep 2024 09:51:46 GMT
< cache-control: public, max-age=2592000
< server: gws
< content-length: 220
< x-xss-protection: 0
< x-frame-options: SAMEORIGIN
< alt-svc: h3=":443"; ma=2592000,h3-29=":443"; ma=2592000
< 
* TLSv1.2 (IN), TLS header, Supplemental data (23):
{ [5 bytes data]
* TLSv1.2 (IN), TLS header, Supplemental data (23):
{ [5 bytes data]
* Connection #0 to host google.com left intact
```
nordvpn ON, google.com curl NOT OK answer
```
curl -v https://google.com -s > /dev/null 
*   Trying 192.0.0.88:443...
* Connected to google.com (192.0.0.88) port 443 (#0)
* ALPN, offering h2
* ALPN, offering http/1.1
*  CAfile: /etc/ssl/certs/ca-certificates.crt
*  CApath: /etc/ssl/certs
* TLSv1.0 (OUT), TLS header, Certificate Status (22):
} [5 bytes data]
* TLSv1.3 (OUT), TLS handshake, Client hello (1):
} [512 bytes data]
* TLSv1.0 (OUT), TLS header, Unknown (21):
} [5 bytes data]
* TLSv1.3 (OUT), TLS alert, decode error (562):
} [2 bytes data]
* error:0A000126:SSL routines::unexpected eof while reading
* Closing connection 0
```
nordvpn OFF, google.com curl OK answer
```
curl -v https://google.com -s > /dev/null 
*   Trying 172.217.168.14:443...
* Connected to google.com (172.217.168.14) port 443 (#0)
* ALPN, offering h2
* ALPN, offering http/1.1
*  CAfile: /etc/ssl/certs/ca-certificates.crt
*  CApath: /etc/ssl/certs
* TLSv1.0 (OUT), TLS header, Certificate Status (22):
} [5 bytes data]
* TLSv1.3 (OUT), TLS handshake, Client hello (1):
} [512 bytes data]
* TLSv1.2 (IN), TLS header, Certificate Status (22):
{ [5 bytes data]
* TLSv1.3 (IN), TLS handshake, Server hello (2):
{ [122 bytes data]
* TLSv1.2 (IN), TLS header, Finished (20):
{ [5 bytes data]
* TLSv1.2 (IN), TLS header, Supplemental data (23):
{ [5 bytes data]
* TLSv1.3 (IN), TLS handshake, Encrypted Extensions (8):
{ [15 bytes data]
* TLSv1.3 (IN), TLS handshake, Certificate (11):
{ [6300 bytes data]
* TLSv1.3 (IN), TLS handshake, CERT verify (15):
{ [79 bytes data]
* TLSv1.3 (IN), TLS handshake, Finished (20):
{ [52 bytes data]
* TLSv1.2 (OUT), TLS header, Finished (20):
} [5 bytes data]
* TLSv1.3 (OUT), TLS change cipher, Change cipher spec (1):
} [1 bytes data]
* TLSv1.2 (OUT), TLS header, Supplemental data (23):
} [5 bytes data]
* TLSv1.3 (OUT), TLS handshake, Finished (20):
} [52 bytes data]
* SSL connection using TLSv1.3 / TLS_AES_256_GCM_SHA384
* ALPN, server accepted to use h2
* Server certificate:
*  subject: CN=*.google.com
*  start date: Jul 30 12:32:53 2024 GMT
*  expire date: Oct 22 12:32:52 2024 GMT
*  subjectAltName: host "google.com" matched cert's "google.com"
*  issuer: C=US; O=Google Trust Services; CN=WR2
*  SSL certificate verify ok.
* Using HTTP2, server supports multiplexing
* Connection state changed (HTTP/2 confirmed)
* Copying HTTP/2 data in stream buffer to connection buffer after upgrade: len=0
* TLSv1.2 (OUT), TLS header, Supplemental data (23):
} [5 bytes data]
* TLSv1.2 (OUT), TLS header, Supplemental data (23):
} [5 bytes data]
* TLSv1.2 (OUT), TLS header, Supplemental data (23):
} [5 bytes data]
* Using Stream ID: 1 (easy handle 0x562fd4dafeb0)
* TLSv1.2 (OUT), TLS header, Supplemental data (23):
} [5 bytes data]
> GET / HTTP/2
> Host: google.com
> user-agent: curl/7.81.0
> accept: */*
> 
* TLSv1.2 (IN), TLS header, Supplemental data (23):
{ [5 bytes data]
* TLSv1.3 (IN), TLS handshake, Newsession Ticket (4):
{ [282 bytes data]
* TLSv1.3 (IN), TLS handshake, Newsession Ticket (4):
{ [282 bytes data]
* old SSL session ID is stale, removing
* TLSv1.2 (IN), TLS header, Supplemental data (23):
{ [5 bytes data]
* TLSv1.2 (OUT), TLS header, Supplemental data (23):
} [5 bytes data]
* TLSv1.2 (IN), TLS header, Supplemental data (23):
{ [5 bytes data]
* TLSv1.2 (IN), TLS header, Supplemental data (23):
{ [5 bytes data]
< HTTP/2 301 
< location: https://www.google.com/
< content-type: text/html; charset=UTF-8
< content-security-policy-report-only: object-src 'none';base-uri 'self';script-src 'nonce-vz2VkIzY-_as4uNVZbHIjA' 'strict-dynamic' 'report-sample' 'unsafe-eval' 'unsafe-inline' https: http:;report-uri https://csp.withgoogle.com/csp/gws/other-hp
< date: Mon, 12 Aug 2024 09:57:24 GMT
< expires: Wed, 11 Sep 2024 09:57:24 GMT
< cache-control: public, max-age=2592000
< server: gws
< content-length: 220
< x-xss-protection: 0
< x-frame-options: SAMEORIGIN
< alt-svc: h3=":443"; ma=2592000,h3-29=":443"; ma=2592000
< 
* TLSv1.2 (IN), TLS header, Supplemental data (23):
{ [5 bytes data]
* TLSv1.2 (IN), TLS header, Supplemental data (23):
{ [5 bytes data]
* Connection #0 to host google.com left intact
```


with nordvon on for  openssl s_client -connect google.com:443 -tls1_2 I sometimes get

```
openssl s_client -connect google.com:443 -tls1_2
CONNECTED(00000003)
40A7CC7C237F0000:error:0A000126:SSL routines:ssl3_read_n:unexpected eof while reading:../ssl/record/rec_layer_s3.c:317:
---
no peer certificate available
---
No client certificate CA names sent
---
SSL handshake has read 0 bytes and written 214 bytes
Verification: OK
---
New, (NONE), Cipher is (NONE)
Secure Renegotiation IS NOT supported
Compression: NONE
Expansion: NONE
No ALPN negotiated
SSL-Session:
    Protocol  : TLSv1.2
    Cipher    : 0000
    Session-ID: 
    Session-ID-ctx: 
    Master-Key: 
    PSK identity: None
    PSK identity hint: None
    SRP username: None
    Start Time: 1723456791
    Timeout   : 7200 (sec)
    Verify return code: 0 (ok)
    Extended master secret: no
---
```

**Comment by tm4ig** [comment]
If I am trying plain http (not https) I am sometimes get this error with enabled nordvpn
``
curl http://google.com -v
*   Trying 192.0.0.88:80...
* Connected to google.com (192.0.0.88) port 80 (#0)
> GET / HTTP/1.1
> Host: google.com
> User-Agent: curl/7.81.0
> Accept: */*
> 
* Mark bundle as not supporting multiuse
< HTTP/1.1 400 Bad request
< content-length: 90
< cache-control: no-cache
< content-type: text/html
< 
<html><body><h1>400 Bad request</h1>
Your browser sent an invalid request.
</body></html>
* Connection #0 to host google.com left intact
```

**Comment by tm4ig** [comment]
I still get errors sometimes with nordvpn
```
curl https://google.com
curl: (35) error:0A000126:SSL routines::unexpected eof while reading
```
or
```
curl http://google.com
<html><body><h1>400 Bad request</h1>
Your browser sent an invalid request.
</body></html>
```

**Comment by mariusSincovici** [comment]
Hi,
Thank you for your patience.

Regarding the custom IP for google: we are using these local IPs as a part of our internal solution, responsible for delivering seamless and uninterrupted browsing experience. These IP addresses are used to proxy certain traffic to the best quality endpoint of our infrastructure for the given website, while keeping the traffic content 100% secure, encrypted, un-tracked and unknown even for our engineers.

For the curl errors: to ensure the solution is effective and reliable, we have implemented an automated abuse prevention mechanism designed specifically to deter bots or scrapers. It not only enhances the authenticity of the traffic but also helps prevent server errors that can occur from overly rapid connections to our endpoints. The errors you received are a result of this mechanism in action. Specifically, the SSL error is caused by our server rejecting the connection due to a pattern indicative of non-human traffic, but the client, expecting a normal SSL response, interprets the absence of this information as an error. During regular browsing, this issue does not occur.

**Comment by tm4ig** [comment]
I have similar error with during regular browsing in firefox.

**Comment by mariusSincovici** [comment]
Hi,
Thank again for your patience.

We identified a bug that caused connection issues, though it only affected abusers. Your browser experience was impacted due to the strict limitations in place in that country to prevent abuse, along with the forced TCP reset. We’ve since released a fix and thoroughly tested the solution. Your feedback is greatly appreciated, and we’d love to offer you additional service time (please check your inbox). If everything looks good, we will close this issue in a week. Don’t hesitate to reach out if anything else comes up.
When you'll have time please check and let us know if it is all working fine.

**Comment by tm4ig** [comment]
There are no any problems with google.com and nordvpn for this week.
I am not abuser, but I connect to nordvpn from my VPS and I connect to VPS via xray reality, because I can not to connect to nordvpn servers directly.
 > Your feedback is greatly appreciated, and we’d love to offer you additional service time (please check your inbox)
What do you mean? There no messages from nordvpn in my inbox

**Comment by mariusSincovici** [comment]
Actually this is something that I wanted to ask.
We would like to send you an email, but we cannot identify your account and cannot send a DM on GitHub.
Do you by any chance have some temporary email address(in case you don't want to make your personal one public) to send to you?
If not I'll think of some other way to send.

**Comment by tm4ig** [comment]
my email and account Qtm4ig@gmail.com

**Comment by mariusSincovici** [comment]
Thanks, I've passed the info and you'll get an email in the upcoming days from us.
Let me know, if you'll not receive by the end of the next week.

**Comment by tm4ig** [comment]
I have taken and applied nordvpn subscribe code from my email.
I think this report can be closed. Thank for your help.

**Comment by ethaniel** [comment]
This happens when you have a nordvpn-inside-nordvpn connection. For example, your router is already connected through nordvpn, but you're also initiating another nordvpn connection from your computer.

**Comment by emmanuelux** [comment]
> Hi, Thank you for your patience.
> 
> Regarding the custom IP for google: we are using these local IPs as a part of our internal solution, responsible for delivering seamless and uninterrupted browsing experience. These IP addresses are used to proxy certain traffic to the best quality endpoint of our infrastructure for the given website, while keeping the traffic content 100% secure, encrypted, un-tracked and unknown even for our engineers.
> 
> For the curl errors: to ensure the solution is effective and reliable, we have implemented an automated abuse prevention mechanism designed specifically to deter bots or scrapers. It not only enhances the authenticity of the traffic but also helps prevent server errors that can occur from overly rapid connections to our endpoints. The errors you received are a result of this mechanism in action. Specifically, the SSL error is caused by our server rejecting the connection due to a pattern indicative of non-human traffic, but the client, expecting a normal SSL response, interprets the absence of this information as an error. During regular browsing, this issue does not occur.

hello thx for explaination, can Nordvpn list all private ip used as proxy as this one for google, is Netflix,  others ??

**Comment by v-ip-er** [comment]
I want to report, in Apr/May 2026, this problem has returned.
I have an OVPN configured at router level (and I have for many years now) for which I quarterly change the OVPN configuration out for an alternate one.

I am in the UK.
I am not a "abuser" I am doing regular browsing.

In the last 5 weeks I have experienced EXACTLY the errors above that @mariusSincovici you suggest were resolved by a fix.

If I turn off my VPN at the router, browsing is fine.
If I turn it on connected with a OVPN downloaded from Nord, I get a failure SPECIFICALLY to google.com and bbc.co.uk while subdomains (e.g. translate.google.com, stage.bbc.co.uk) resolve fine, as do many many other websites.

If I try to ping bbc.co.uk, I get exactly the response noted above, I get a 192.0.0.60. For google.com I get 192.0.0.88.

I have linux devices on my network, if I do a dig on them for bbc.co.uk I get back the correct public facing IPs.

What's the resolution here?, because I have a paid for Nord account which is, essentially, preventing perfectly acceptable browsing.

**Comment by DaveRune** [comment]
> I want to report, in Apr/May 2026, this problem has returned. I have an OVPN configured at router level (and I have for many years 
> ...

It took me a while to find this issue. But what you described is exactly my issue with identical setup.

**Comment by emmanuelux** [comment]
with pf on openbsd you can make a rule to bypass this behavior

**Comment by DaveRune** [comment]
Fixed it on OpenWRT. SSH in and run ```vi /etc/config/dhcp```
Replace:
```
list server '103.86.96.100'
list server '103.86.99.100'
```
with
```
list server '1.1.1.1'
list server '1.0.0.1'
```
This sets the DNS servers to cloudflare.
Use escape and type ```:wq``` to save and exit

Then run ```/etc/init.d/dnsmasq restart```

@v-ip-er Hopefully this helps you. 🤞

**edit:** actually I don't think this is a great idea. Other sites are now not working. Can't even access nordvpns account or support pages. Tried 8.8.8.8 and 8.8.4.4. No luck

**Comment by v-ip-er** [comment]
Thank for the suggestion @DaveRune, I couldn't do this because my router is stock and I don't have SSH capability to the router (same reason I can't use your suggestion @emmanuelux).

Equally I wouldn't change the DNS path because changing the routing path to use public DNS like cloudflare (or even something like 9.9.9.9 through Quad9) would create DNS leakage which is of course exactly why I have a VPN in the first place.

Annoyingly I can report a further failure since day before yesterday (I haven't changed my config, VPN is still [on] at router level).

Now.... all my Amazon IoT devices are flashing orange and Alexa no longer responds in my house (a much bigger problem than some browsing I'm sure you'll agree @mariusSincovici) this is likely because some part of the Amazon/AWS back-end is also blocked by the same thing as google and bbc. I know this because, I cannot even browse to the Amazon.co.uk website, and ping returns a reserved bogon range, which is not publicly routable, the IP of which is not part of Amazon's public IP ranges. 

> ping amazon.co.uk
> 
> Pinging amazon.co.uk [198.18.2.14] with 32 bytes of data:
> Request timed out.
> Request timed out.
> Request timed out.
> Request timed out.
> 
> Ping statistics for 198.18.2.14:
>     Packets: Sent = 4, Received = 0, Lost = 4 (100% loss),

**Comment by emmanuelux** [comment]
you coud use unbound on the vpn interface to replace nordvpn dns.

**Comment by v-ip-er** [comment]
Thanks for the Unbound suggestion @emmanuelux.

I can see why that approach works for sure,
It's not viable for my setup for two sticky reasons:

1) is DNS leak risk: I'm using the VPN specifically to keep all DNS traffic inside the encrypted tunnel (obv). So routing queries via Unbound to external resolvers, even with DoT, would defeat the purpose and potentially expose DNS lookups outside the tunnel. That's a dealbreaker for me. I may as well just turn the VPN off at that point which would solve this.

2) I'm limited because I have stock firmware on my router (no OPNsense/pfSense/OpenWRT) so I don't have the interface-level control to bind Unbound to the VPN interface properly, and I'm trying hard to avoid flashing custom firmware because I'd have to manually rebuild a ton of internal network configuration.
_I know I could just put Unbound on a Pi, but then I'm back to point (1)_

Appreciate the idea though. And happy to receive any other assistance anyone might have (👋 @mariusSincovici )

The root cause is confirmed to be NordVPN's in-tunnel DNS filtering returning a sinkhole for legitimate domains;

```
PS C:\> Resolve-DnsName bbc.co.uk -Server 103.86.96.100

Name                                           Type   TTL   Section    IPAddress
----                                           ----   ---   -------    ---------
bbc.co.uk                                      A      60    Answer     192.0.0.60


PS C:\> Resolve-DnsName google.com -Server 103.86.96.100

Name                                           Type   TTL   Section    IPAddress
----                                           ----   ---   -------    ---------
google.com                                     A      60    Answer     192.0.0.88


PS C:\> Resolve-DnsName bbc.co.uk -Server 1.1.1.1

Name                                           Type   TTL   Section    IPAddress
----                                           ----   ---   -------    ---------
bbc.co.uk                                      AAAA   51    Answer     2a04:4e42::81
bbc.co.uk                                      AAAA   51    Answer     2a04:4e42:600::81
bbc.co.uk                                      AAAA   51    Answer     2a04:4e42:200::81
bbc.co.uk                                      AAAA   51    Answer     2a04:4e42:400::81
bbc.co.uk                                      A      293   Answer     151.101.0.81
bbc.co.uk                                      A      293   Answer     151.101.192.81
bbc.co.uk                                      A      293   Answer     151.101.64.81
bbc.co.uk                                      A      293   Answer     151.101.128.81


PS C:\> Resolve-DnsName google.com -Server 9.9.9.9

Name                                           Type   TTL   Section    IPAddress
----                                           ----   ---   -------    ---------
google.com                                     AAAA   150   Answer     2a00:1450:4001:c15::66
google.com                                     AAAA   150   Answer     2a00:1450:4001:c15::8b
google.com                                     AAAA   150   Answer     2a00:1450:4001:c15::8a
google.com                                     AAAA   150   Answer     2a00:1450:4001:c15::65
google.com                                     A      279   Answer     142.251.14.139
google.com                                     A      279   Answer     142.251.14.101
google.com                                     A      279   Answer     142.251.14.113
google.com                                     A      279   Answer     142.251.14.100
google.com                                     A      279   Answer     142.251.14.138
google.com                                     A      279   Answer     142.251.14.102
```

I'm pursuing it with Nord support team but they are running me around the houses at the moment spent hours on chat, and they are giving me standard troubelshooting that will never work. So I have to hurdle this first level support nonsense first before an ACTUAL tech will look at this and see the problem.

/Cheers

**Comment by drm20000** [comment]
I confirm that I am having the same issue now.  When Nord resolves google.com it returns 192.0.0.88 which then does nothing. 

If I choose to not use nord’s dns servers, there is no problem.  If I use nords dns server thru the vpn it fails

**Comment by emmanuelux** [comment]
why use nordvpn dns server?  you could use your owns

**Comment by drm20000** [comment]
> why use nordvpn dns server? you could use your owns

Because I want my DNS Query to go through the VPN. Why should I switch? I am paying for Nord and their VPN with DNS service?

**Comment by emmanuelux** [comment]
so unbound over vpn,  no pb

**Comment by illusionmanager** [comment]
I'm having the same issue, but I don't think it is nordvpn's problem.
I got a brand new router (GL MT-6000 flint2) and I configured nordvpn inside the router. It works great except for google.com.
On windows it tells me the address is 192.0.0.88. 
But I investigated a bit further by going inside the router using
ssh root@192.168.178.2 (my routers address)

When I check the dns result via Nordvpn's dns server using
dig @103.86.96.100 google.com I get the correct result.
also 
dig @127.0.0.1 google.com    works fine
However
dig @127.0.0.1 -p 4153 google.com         -> 192.0.0.88

and that is the port used with  dnsmasq.ovpnclient1 on port 4153

It still is the stock firmware and there might be some Chinese malware on it, so perhaps I should return it.

**Comment by v-ip-er** [comment]
It's 100% a Nord problem. If I have the VPN [on] my browsing and DNS is broken, if I turn it [off] all my functionality returns. 
There really doesn't need to be any further testing than that.

Of course, WHERE the problem is in Nord is a different question.

I have the same experience.

[1] If I resolve DNS with my VPN [on] and no DNS server selected I get the bogon IPs ❌ 
[2] If I do the same test using the Quad9 or Cloud flare DNS, I get the correct IPs ✅ 
[3] If I manually resolve DNS using the two published DNS servers from Nord ✅ 

The reality is, we have absolutely no idea what the servers that my OVPN service is connecting to as this is not transparent or published. It's clearly not these two Nord DNS servers or there would be no problem.

Further, my network is very secure, there are no devices I don't know about on my network. I have had my router for several YEARS and I have done nothing on it (including I haven't flashed it with any third-party firmware like Merlin) except change the OVPN configuration server from time to time.

I've tried 9 different servers directly downloaded config files from Nord OVPN downloads.
They are all banjaxed.

**Comment by illusionmanager** [comment]
well, if you ask the nordvpn dns server directly (from within the router) for the answer, it is correct, if you ask it through the program the router uses internally the answer is wrong. Is it the program or the server that is wrong??
So my conclusion would be it is a VPN problem inside the router as nordvpn behaves as it should. 

Direct query from router to 103.86.99.100 gives valid Google IP.
The same query from a LAN client to 103.86.99.100 gives 192.0.0.88.
Therefore LAN DNS interception/dispatcher is modifying the result.

As a workaround, but definitely not a solution, i added 5 hosts to /etc/hosts inside the router
142.250.180.14 google.com
142.250.180.14 www.google.com
142.250.180.14 connectivitycheck.gstatic.com
142.250.180.14 connectivitycheck.android.com
142.250.180.14 clients3.google.com
(the last 3 are to make my phone happy)

**Comment by drm20000** [comment]
> so unbound over vpn, no pb

Well unbound does not encrypt the dns traffic unless you route it through a vpn.

Does not really work for me.  My router allows me to have a failover VPN connection. So if Nord connection 1 goes down then it auto switches to Nord connection 2. This works fine.

So I tried unbound (it is installed on my router) and routed the unbound traffic through a Nord vpn connection. But there is no failover on my router for this traffic. So when the Nord connection fails, unbound fails. I find it not uncommon for a Nord connection to go down temporarily a couple of times per day … and of course that is when you need a connection

The best solution is for nord to fix the problem.  This is a recent problem for me.

**Comment by DoYouEvenCpp** [comment]
Hey @v-ip-er ,

We apologise for the delay in responding to this thread. I've reopened the issue so all notifications are now clearly visible.

In addition to the logs you've already provided, could you also collect the nordvpnd service logs by running the following command: `sudo journalctl -u nordvpnd.service > daemon.log`. We'd also appreciate it if you could tell us which specific server you're connected to when the issue occurs – you can check this by running `nordvpn status` once you're connected to the VPN.

If you have any privacy concerns about sharing the logs here, please feel free to submit this information through our customer care channels, and do reference this issue so we can identify it properly.

**Comment by illusionmanager** [comment]
although I initially thought my new router was to blame, it is nordvpn that is to blame
I just did a tcpdump from inside the router, it clearly shows nordvpn is sending the wrong answer back
12:05:22.244500 IP 10.100.0.2.52846 > 103.86.96.100.53: 57155+ [1au] A? google.com. (51)
12:05:22.244545 IP 10.100.0.2.52846 > 103.86.99.100.53: 57155+ [1au] A? google.com. (51)
12:05:22.255511 IP 103.86.96.100.53 > 10.100.0.2.52846: 57155 1/0/1 A 192.0.0.88 (55)
12:05:22.267838 IP 103.86.99.100.53 > 10.100.0.2.52846: 57155 1/0/1 A 192.0.0.88 (55)

**Comment by DoYouEvenCpp** [comment]
I see. @illusionmanager  could you also provide the same information, such as the daemon logs, the server you're connected to, and which version of the NordVPN App you're running? The commands you can use are listed in my previous response.

**Comment by illusionmanager** [comment]
This is occurring on a GL.iNet MT6000 router running OpenWrt/GL.iNet firmware, not on a Linux PC using the NordVPN application.

The router does not use systemd and therefore has no journalctl or nordvpnd.service.

The VPN connection is configured directly in the router firmware using NordVPN DNS servers 103.86.96.100 and 103.86.99.100.

It doesn't matter which nordvpn node I connect to, i tried several, and the result is the same

**Comment by illusionmanager** [comment]
could it have to do with me discarding two older asus routers with nordvpn running on them, and just replacing it with a newer router?? What I mean is perhaps too many installs of nordvpn?

**Comment by DaveRune** [comment]
> could it have to do with me discarding two older asus routers with nordvpn running on them, and just replacing it with a newer router?? What I mean is perhaps too many installs of nordvpn?

No @illusionmanager that wont be the case as it's not just affecting you.
This issue is still present for me, my setup had been running fine until a few weeks ago. No change to my setup in months prior to that. I'm on cloudflare DNS right now and I'd love to get off it, as v-ip-er says, not much point having a VPN right now.

> The VPN connection is configured directly in the router firmware using NordVPN DNS servers 103.86.96.100 and 103.86.99.100. It doesn't matter which nordvpn node I connect to, i tried several, and the result is the same

Identical scenario for me too.

**Comment by mariusSincovici** [comment]
Hi,

Thank for your feedback, first if you could help us to better understand the problems, it would be very helpful. 
And if needed we'll split in separate issues to better keep track, because this issue is getting quite big.

1. The initial issues was because of `curl https://google.com -v`. At that time the problem was found and fixed. Is it something similar that happens to you @v-ip-er, but this time with `ping amazon.co.uk`?
 
2. There are questions about the wrong IPs returned for different domains, while connected to VPN and using NordVPN DNS servers. This is intended and it must not affect the functionality, if it does then there are issues that we need to fix.

Let me know if I've missed something.

**Comment by illusionmanager** [comment]
can I ask "There are questions about the wrong IPs returned for different domains, while connected to VPN and using NordVPN DNS servers. This is intended and it must not affect the functionality, if it does then there are issues that we need to fix"

Why would you return 192.0.0.88 for google.com to me??  If that is intended, what would trigger that?? Surely returning 192.0.0.88 for google does affect functionality... strongly!
I'm just a causal user from the netherlands. How do I prevent this from triggering it? I tried server from germany, and even from singapore, they all do the same.

**Comment by emmanuelux** [comment]
it is probably a nordvpn transparent proxy, you could nat 192.0.0.88 to vpn interface to get it work, cause you probably not nat all ip 192.x.x.x (private ip)

**Comment by mariusSincovici** [comment]
This translation takes place when using NordVPN DNS servers. The DNS request is sent to our DNS servers and the domain is resolved to that IP address. This only happens when the request is sent thru VPN, not outside.

This was implemented to make the VPN usage less problematic. For example Google may show CAPTCHAs when using a VPN. Others can block completely the service and so on.

To not use this would mean that you'll need to use some DNS servers other than the default ones from the application. For example 1.1.1.1 or 8.8.8.8, `nordvpn set dns 1.1.1.1`. Then the real IP address is returned instead. The DNS requests still go thru the tunnel, but to a 3rd party server.

Is google.com not working at all for you or just sometimes?

**Comment by drm20000** [comment]
As I read through the comments, I see this problem on gl-inet ans asus routers.  My router is a Firewalla and has the same issue of it returning 192.0.0.8 for google.com.  I do not have the same problem if I disable NordVPN on my router and use the Nord client on my iphone or ipad

All 3 of these routers are running a form of linux .. so this seems to be a common thread.  Correct me if I am wrong!

**Comment by illusionmanager** [comment]
@mariusSincovici I don't understand this at all. 
When I set dns to use 8.8.8.8 or 1.1.1.1 it works fine, but I'm supposed to be using nordvpn's dns address. When I do that, i cannot reach www.google.com at all. 
How would it know the right address for google if nordvpn gives me the wrong answer?

**Comment by mariusSincovici** [comment]
When you're using the VPN application on your desktop, then the entire traffic is sent thru the VPN tunnel because it is the default routing interface. In that case it doesn't matter which is the IP address of google as long as the VPN server knows what to do with it.

In case of google think of VPN server, as @emmanuelux mention, like a proxy server that takes your requests and sends them to google, then back to you.

You can also check other platforms after you're connected to VPN with default config, google will be resolved to 192.0.0.8.

Because this seams to be a different issue from the original one and there are more who have this issue, I'll create tomorrow a separate ticket to see how could this be addressed, so it will help also others in the future.

In the meantime, could you check what happens on the router with the traffic to/from google? What interface is being used to send the traffic to 192.0.0.8? As @emmanuelux was suggesting a possible solution is to NAT to vpn interface, for now or just use the custom DNS.

**Comment by emmanuelux** [comment]
on openbsd I nat destination to 192.0.0.88 to the vpn interface and it is ok. else unbound to avoid dns of nordvpn. what trick nordvpn uses to avoid captcha with the proxy?

**Comment by illusionmanager** [comment]
i don't know how to " NAT to vpn interface". Could you tell me the command of how to do that?

**Comment by v-ip-er** [comment]
@DoYouEvenCpp @mariusSincovici

Thanks for reopening this and for the continued attention.

Unfortunately I can't provide the exact information you've requested due to a the technical constraints on my side.

- I cannot provide journalctl or nordvpnd.service logs as I am not running any NordVPN application. My connection is configured through my ASUS router which has stock firmware (no Merlin or flashed firmware). As such, the router does not have an accessible Linux shell, systemd, or a safely enabled SSH capability—enabling it on stock firmware is a bit complex and high-risk that could brick my router which obviously I'd like to avoid.
As such, I cannot run the daemon-level commands you've asked for.

Some additional information for your knowledge;

1. I have an open case for this with Nord (20893427) which has been escalated from first-level support after a bit of back-and-forth but has gone, nowhere for many days. It should be noted that in this ticket I detailed information as to my troubleshooting in an email Tue 05/05/2026 AND I also sent several attachments in a chat, including an excerpt of my router log where you can see the VPN going from [offline] to [online].

2. The issue is growing and now breaks basic functionality. As reported by others, this is not just about bbc.co.uk or google.com resolving to 192.0.0.x addresses. The problem is expanding and now actively blocks amazon.co.uk, which has completely disabled my Alexa-enabled smart home devices and remote access. As @DaveRune perfectly stated, this makes the VPN effectively pointless right now. I also tried to use Claude.ai yesterday and it too is now unroutable with the router VPN client enabled.

3) Any "NAT" or "custom DNS" workarounds are not practical. The discussion between @emmanuelux and @mariusSincovici about NATing 192.0.0.88 to the VPN interface or using a custom DNS like 1.1.1.1 is understood, but it's not a feasible workaround on consumer stock router firmware. I lack the low-level interface control. 
Equally, using a third-party DNS, has the high potential (if not certainty) of leaking requests outside of NordVPN's infrastructure, defeating the privacy purpose of using the VPN in the first place. 
Put simply, we are paying for a service that should handle its own DNS securely.

Here's a bit of the basic detail so you can see what's happening with basic DNS resolution.

> PS C:\> Resolve-DnsName google.com -Server 103.86.96.100
> 
> Name                                           Type   TTL   Section    IPAddress
> google.com                                     A      60    Answer     192.0.0.88
> 
> ----                                           ----   ---   -------    ---------
> PS C:\> Resolve-DnsName google.com -Server 103.86.99.100
> 
> Name                                           Type   TTL   Section    IPAddress
> google.com                                     A      60    Answer     192.0.0.88
> 
> ----                                           ----   ---   -------    ---------
> PS C:\> Resolve-DnsName google.com -Server 9.9.9.9
> 
> Name                                           Type   TTL   Section    IPAddress
> google.com                                     AAAA   250   Answer     2a00:1450:4001:c13::64
> google.com                                     AAAA   250   Answer     2a00:1450:4001:c13::66
> google.com                                     AAAA   250   Answer     2a00:1450:4001:c13::8a
> google.com                                     AAAA   250   Answer     2a00:1450:4001:c13::65
> google.com                                     A      202   Answer     142.251.13.100
> google.com                                     A      202   Answer     142.251.13.138
> google.com                                     A      202   Answer     142.251.13.101
> google.com                                     A      202   Answer     142.251.13.102
> google.com                                     A      202   Answer     142.251.13.113
> google.com                                     A      202   Answer     142.251.13.139
> 
> ----                                           ----   ---   -------    ---------
> PS C:\> Resolve-DnsName amazon.co.uk -Server 103.86.99.100
> 
> Name                                           Type   TTL   Section    IPAddress
> amazon.co.uk                                   A      60    Answer     198.18.2.14
> 
> ----                                           ----   ---   -------    ---------
> PS C:\> Resolve-DnsName amazon.co.uk -Server 1.1.1.1
> 
> Name                                           Type   TTL   Section    IPAddress
> amazon.co.uk                                   A      13    Answer     3.254.236.223
> amazon.co.uk                                   A      13    Answer     3.253.170.23
> amazon.co.uk                                   A      13    Answer     3.253.181.192
> 
> ----                                           ----   ---   -------    ---------
> PS C:\> Resolve-DnsName bbc.co.uk -Server 103.86.96.100
> 
> Name                                           Type   TTL   Section    IPAddress
> bbc.co.uk                                      A      60    Answer     192.0.0.60
> 
> ----                                           ----   ---   -------    ---------
> PS C:\> Resolve-DnsName bbc.co.uk -Server 8.8.8.8
> 
> Name                                           Type   TTL   Section    IPAddress
> bbc.co.uk                                      AAAA   145   Answer     2a04:4e42:400::81
> bbc.co.uk                                      AAAA   145   Answer     2a04:4e42::81
> bbc.co.uk                                      AAAA   145   Answer     2a04:4e42:200::81
> bbc.co.uk                                      AAAA   145   Answer     2a04:4e42:600::81
> bbc.co.uk                                      A      158   Answer     151.101.0.81
> bbc.co.uk                                      A      158   Answer     151.101.128.81
> bbc.co.uk                                      A      158   Answer     151.101.192.81
> bbc.co.uk                                      A      158   Answer     151.101.64.81

Here's the same test with my VPN turned [on] at router level:

> PS C:\> Resolve-DnsName bbc.co.uk
> 
> Name                                           Type   TTL   Section    IPAddress
> bbc.co.uk                                      A      60    Answer     192.0.0.60
> 
> ----                                           ----   ---   -------    ---------
> PS C:\> Resolve-DnsName google.com
> 
> Name                                           Type   TTL   Section    IPAddress
> google.com                                     A      60    Answer     192.0.0.88
> 
> ----                                           ----   ---   -------    ---------
> PS C:\> Resolve-DnsName amazon.co.uk
> 
> Name                                           Type   TTL   Section    IPAddress
> amazon.co.uk                                   A      60    Answer     198.18.2.14

It's becoming clear that this is a server-side issue affecting users who connect via router-level OpenVPN configurations using NordVPN's default DNS servers. You asked **which** servers I had tried - for the last 6 or so weeks, here is the answer (I tried both UDP and TCP configuration files):

> **TCP**
> uk2573.nordvpn.com.tcp.ovpn
> uk2578.nordvpn.com.tcp.ovpn
> uk2954.nordvpn.com.tcp.ovpn
> uk2955.nordvpn.com.tcp.ovpn
> 
> **UDP**
> uk2182.nordvpn.com.udp.ovpn
> uk2188.nordvpn.com.udp.ovpn
> uk2190.nordvpn.com.udp.ovpn
> uk2191.nordvpn.com.udp.ovpn
> uk2192.nordvpn.com.udp.ovpn
> uk2573.nordvpn.com.udp.ovpn
> uk2745.nordvpn.com.udp.ovpn
> uk2827.nordvpn.com.udp.ovpn

The "intended" DNS translation your team described is currently broken for a specific set of configurations per the above.

_I think_ the variable that needs to be checked is the filtering policy on your DNS servers for these domains when requests come through the VPN tunnel. This feels easy enough to replicate also.

I appreciate the offer to create a separate ticket. If you **need** to do this, then please link me to it - though I have provided my ticket reference above. I'll do my best to provide any additional information needed. However I think it's very clear from @DaveRune and my examples alone that this problem is in the Nord VPN tunnel.

**Comment by mariusSincovici** [comment]
Thanks all for your info and feedback.

@v-ip-er I've found the customer support ticket mentioned.
@emmanuelux I don't know how is implemented on the backend side. 😄 

Since the issue is not caused by the NordVPN Linux app, we first need to discuss with the backend teams about it. As soon as we have any new info we'll let you know.

**Comment by drm20000** [comment]
As others have noted the problem is not limited to google.com
I am using a Firewalla router and the Nord VPN connection is enabled on the router
I am connected to NORD us.9392 server using openVPN tcp protocol.  I have enabled DNS over VPN

google.com returns an IP address of 192.0.0.8
amazon.com returns an IP address of 198.18.2.14 often.

Both addresses are not public and neither address is reachable. 
If i disable DNS over VPN, then both sites resolve properly and are reachable. Amazon will sometime fail with DNS over VPN enabled. But it always works if I disable DNS over VPN

**Comment by v-ip-er** [comment]
@DaveRune, @drm20000, 

I got an email from support re: my ticket, which reads...

> We have made some changes on our end, and the issue is no longer reproducible.
> Could you please try connecting to a few different locations/servers and let us know if the issue persists?
> We would appreciate your feedback after testing. Take care!

I did some basic testing...

```
PS C:\> Resolve-DnsName google.com -Server 103.86.96.100

Name                                           Type   TTL   Section    IPAddress
----                                           ----   ---   -------    ---------
google.com                                     AAAA   98    Answer     2a00:1450:4007:810::200e
google.com                                     A      38    Answer     142.251.39.110


PS C:\> Resolve-DnsName bbc.co.uk -Server 103.86.96.100

Name                                           Type   TTL   Section    IPAddress
----                                           ----   ---   -------    ---------
bbc.co.uk                                      AAAA   269   Answer     2a04:4e42:600::81
bbc.co.uk                                      AAAA   269   Answer     2a04:4e42:400::81
bbc.co.uk                                      AAAA   269   Answer     2a04:4e42::81
bbc.co.uk                                      AAAA   269   Answer     2a04:4e42:200::81
bbc.co.uk                                      A      272   Answer     151.101.0.81
bbc.co.uk                                      A      272   Answer     151.101.128.81
bbc.co.uk                                      A      272   Answer     151.101.64.81
bbc.co.uk                                      A      272   Answer     151.101.192.81


PS C:\> Resolve-DnsName amazon.co.uk -Server 103.86.99.100

Name                                           Type   TTL   Section    IPAddress
----                                           ----   ---   -------    ---------
amazon.co.uk                                   A      12    Answer     3.254.236.223
amazon.co.uk                                   A      12    Answer     3.253.181.192
amazon.co.uk                                   A      12    Answer     3.253.170.23
```

I'm gonna turn back on my router VPN and do some more testing.

Do you guys wanna test yours too and see if you're getting the same results before I go back to them?

**Comment by illusionmanager** [comment]
With support I'm still nowhere. From the beginning I contacted them, but they keep asking me to do totally irrelevant things, that - if anything - would only work on windows and not solve the problem on my phone at all and do ignore the problem seen directly from the router. I think the problem is that it isn't the same person handling the support request but every time a different one, that just shouts anything that jumps to his mind, before reading the emails.

Anyway Resolve-DnsName google.com -server 103.86.99.100 still gives the answer 192.0.0.88, nslookup gives the same answer. 

But it looks like Nordvpn's synthetic 192.0.0.88 endpoint is now pingable and it even answers on https port 443.

My browser now can connect to google.com just fine.

So yes, they seemed have solved it.

**Comment by drm20000** [comment]
I also did some testing. 

<img width="1536" height="1125" alt="Image" src="https://github.com/user-attachments/assets/9287c595-ccdb-4053-9bc1-3f3834e1b5ab" />

<img width="1536" height="1365" alt="Image" src="https://github.com/user-attachments/assets/ed16d8bb-c08f-46b6-ab40-630195599ecb" />

So both google.com and amazon.com fails the nslookup command when DNS over VPN is enabled.  With DNS over VPN disabled Google resolves correctly.  Amazon ip4 resolves correctly but ip6 still fails.

However both google ans amazon sites now load properly in my browser.  I do not understand how they load in the browser properly if the ip address returned is not correct.  Can someone explain?

**Comment by illusionmanager** [comment]
@drm20000 I think the idea is that nordvpn sends your dns request the answer 192.0.0.88 on purpose. Your browser thinks that is the address of google and tries to contact it. However that goes through the vpn connection to Nordvpn and nordvpn catches that and forwards the request to google anyway. 
They said to prevent repeated re-captchas etc. 

What I'm curious about is, what they use as ip address to contact google to do that. I surely hope it isn't my ip address, and it probably isn't the ip address of the nordvpn server as I assume that is what triggers the re-captchas.

So that's my question towards Nordvpn: what is the address google thinks the request in coming from?

**Comment by drm20000** [comment]
Thanks for the reply.  But, I am still thinking:

1) my browser asks my dns resolver for the ip address of google.com. 
2) my router processes the request and sends the request to nord.
3) nord replies 192.0.0.88 
4) my router reports “server not found nxdomain”

So doesn’t my browser stop at this point as it does not know where to go?

Or are you saying NORD is sending a reply for google even though the browser has never requested the google IP? In a way, this feels like a “man in the middle” interjecting in my traffic

**Comment by illusionmanager** [comment]
I think nslookup tells you that the ipv4 address of google is 192.0.0.88, but , and I'm not sure, fails to find its ipv6 address. Now it "knows" the address is 192.0.0.88 it will use that. 
Nordvpn will handle it for you.

**Comment by emmanuelux** [comment]
it is probably a transparent proxy to avoid captcha, it works may be by routing to ipv6 with 4-to-6 , google sees then different ip even if it have the same ipv6 prefix.

**Comment by illusionmanager** [comment]
@drm20000 and yes, Nordvpn **is** the man in the middle, all your traffic is going through them. They know which sites you are using (but because https is encrypted they cannot see what the contents is). You have to trust some DNS provider

**Comment by drm20000** [comment]
> I think nslookup tells you that the ipv4 address of google is 192.0.0.88, but , and I'm not sure, fails to find its ipv6 address. Now it "knows" the address is 192.0.0.88 it will use that. Nordvpn will handle it for you.

Well obviously you are correct. I just still have some problems to understand how it is working. i have some more thinking/investigations to do.  Also, I would think that the problem was not limited to google and amazon … so if the nord solution is website specific, we still may encounter additional problems

**Comment by illusionmanager** [comment]
@drm20000 I think that Nordvpn only implemented this for some high-traffic websites that would otherwise would get annoying. As many users use the same nordvpn server, a website might think it is being visited by a robot because it get many page requests coming from the same server. If a website thinks this is a problem it might ask you to verify you are not a robot, but that depends on the website itself. 
Nordvpn uses a trick so when you want to contact google via one of nordvpn's servers, it doesn't use the same server address all the time. @emmanuelux suggests it might convert ipv4 address to ipv6 address. (but I don't know how that helps)

**Comment by v-ip-er** [comment]
After testing by simply re-activating my router-level OVPN config using the server it was already configured to, browsing is working normally again for Google, BBC, Amazon, and other previously blocked domains are all back. Also my IoT devices are also still online now after ~10hrs. DNS leak tests confirm all queries remain within Nord's infrastructure - no leaks.

Whatever server-side change they made seems to have stuck.

@mariusSincovici - it would be good if you can validate this with the technical teams that they have resolved this problem as I imagine it was something that was quite widespread and tell us what occurred if you can.

------------------
Meanwhile, I wanted to share what I think is happening, because the DNS behaviour still looks broken if you only test with command-line tools, and I know that confused me too.

I am getting some slightly odd DNS lookups per below:

```
PS C:\> Resolve-DnsName google.com

Name                                           Type   TTL   Section    IPAddress
----                                           ----   ---   -------    ---------
google.com                                     A      60    Answer     192.0.0.88


PS C:\> Resolve-DnsName bbc.co.uk
Resolve-DnsName : bbc.co.uk : DNS name does not exist
At line:1 char:1
+ Resolve-DnsName bbc.co.uk
+ ~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ResourceUnavailable: (bbc.co.uk:String) [Resolve-DnsName], Win32Exception
    + FullyQualifiedErrorId : DNS_ERROR_RCODE_NAME_ERROR,Microsoft.DnsClient.Commands.ResolveDnsName

PS C:\> Resolve-DnsName amazon.co.uk
Resolve-DnsName : amazon.co.uk : DNS name does not exist
At line:1 char:1
+ Resolve-DnsName amazon.co.uk
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ResourceUnavailable: (amazon.co.uk:String) [Resolve-DnsName], Win32Exception
    + FullyQualifiedErrorId : DNS_ERROR_RCODE_NAME_ERROR,Microsoft.DnsClient.Commands.ResolveDnsName

PS C:\> Resolve-DnsName amazon.co.uk -server 9.9.9.9

Name                                           Type   TTL   Section    IPAddress
----                                           ----   ---   -------    ---------
amazon.co.uk                                   A      60    Answer     3.253.170.23
amazon.co.uk                                   A      60    Answer     3.254.236.223
amazon.co.uk                                   A      60    Answer     3.253.181.192


PS C:\> Resolve-DnsName bbc.co.uk -server 8.8.8.8

Name                                           Type   TTL   Section    IPAddress
----                                           ----   ---   -------    ---------
bbc.co.uk                                      AAAA   183   Answer     2a04:4e42:400::81
bbc.co.uk                                      AAAA   183   Answer     2a04:4e42:600::81
bbc.co.uk                                      AAAA   183   Answer     2a04:4e42:200::81
bbc.co.uk                                      AAAA   183   Answer     2a04:4e42::81
bbc.co.uk                                      A      75    Answer     151.101.64.81
bbc.co.uk                                      A      75    Answer     151.101.0.81
bbc.co.uk                                      A      75    Answer     151.101.192.81
bbc.co.uk                                      A      75    Answer     151.101.128.81
```

When I ran curl -sI https://www.google.com, I got back a completely normal set of Google response headers:

```
-     HTTP/1.1 200 OK – the connection worked.
-     Standard Google headers for content type, caching, and security.
-     Two cookies that Google sets for every visitor to manage bot detection
these are anonymous and contain no personal information.
-     A timestamp showing when Google processed the request.
```

> There is nothing in this response that identifies me, my real IP address, or my device. It's exactly what you'd expect to see from a healthy connection routed through a working transparent proxy.

I believe Nord has implemented a transparent proxy at that sinkhole address. Nord have documented their use of transparent proxies before. In a blog post explaining how their Threat Protection feature works, they describe a transparent proxy as something that sits between our devices and the internet, intercepting traffic invisibly to filter or redirect it . As it's transparent, this is likely why nslookup faithfully reports 192.0.0.88 while our browsing works fine.

My best understanding is that Nord's DNS deliberately returns 192.0.0.88 for big-boy domains like Google and Amazon. We send a request, Nord's proxy catches it, and forwards it to the real destination - probably cycling through different egress IPs to dodge the CAPTCHA triggers which have been a tell-tale sign for me in the past that it's time to change Nord server.
The proxy acts as a middleman that makes traffic look more like a normal user

It's conjecture of course, but it would explains why Resolve-DnsName and nslookup appear to "fail" or return odd results. They're doing exactly what they're supposed to do - reporting the literal DNS response - they don't know the address is a proxy. 
In some cases, I assume the VPN interface is blocking or filtering certain direct DNS requests at the command-line level, which is why some of us saw "DNS name does not exist" errors while browsers worked perfectly.

The browser uses the system resolver, which takes the 192.0.0.88 answer, routes it through the tunnel, and the proxy handles the rest.

I noticed some IPv6-related questions in the thread, and I'll be honest - I haven't got a scooby how Nord's proxy handles the IPv4-to-IPv6 mapping, or even **if** it is. The suggestion around 4-to-6 translation sounds plausible, but I dug and couldn't find anything to confirm it. **If someone from Nord wants to clarify**, I'm sure we'd all be interested.

I guess, in the short window of testing I have done, Nord appears to have fixed whatever was broken in their transparent proxy infrastructure. The 192.0.0.88 address is no longer a sinkhole - it seems to be an active relay. Command-line reporting the proxy address rather than the real destination IP seem to be expected behaviour for a transparent proxy setup 

Everything now works as it should for actual browsing and connectivity, which is what matters.

Thanks to everyone who contributed. I think that our combined evidence made it clear the problem was server-side from the start which may have sped up a resolution.

I'll be keeping an eye on my side, but interested to hear how everyone else's testing gets on.

Cheers all 👍

**Comment by DoYouEvenCpp** [comment]
@v-ip-er  - indeed, the backend team acknowledged the bug and reported that the fix has been deployed (the issue was in the configuration rules they applied to the servers, so it might have affected multiple nodes).

We'll keep the issue open for a while to monitor the situation. Thanks to everyone for your feedback and patience while we worked on this fix.

**Comment by DaveRune** [comment]
I'll update my DNS back to the NordVPN IPs and feed back in the next few days. Thanks for your work and keeping the issue open

**Comment by DaveRune** [comment]
@DoYouEvenCpp 

Update: **Google is fixed**. 

Interestingly though I get: 
```
This site can’t be reached
nordvpn.com’s DNS address could not be found. Diagnosing the problem.
DNS_PROBE_POSSIBLE
```

The above is true for every VPN website I have visited. DNS address could not be found for any of them. @v-ip-er Do you experience this too?
<img width="2704" height="41" alt="Image" src="https://github.com/user-attachments/assets/900d1b58-e227-4818-80b0-209fdf2de61f" />
All other sites I've tried seem fine.
Same results across multiple different servers.

**Comment by v-ip-er** [comment]
@DaveRune, what test are you running to get that _"site can't be reached"_, I'll happily run the same test on my side.

Interesting about 30 minutes ago, my ability to browse to Amazon fell over.
Works on my 4G, doesn't work with my VPN on.

@DoYouEvenCpp, @mariusSincovici - heads up re: Amazon.

**Comment by DaveRune** [comment]
@v-ip-er Just in browser, I noticed that I couldn't access nordvpn.com but I could other sites. Out of curiosity I tried other VPN sites and they didn't work. I found a list of VPNs and tried them all, all blocked.

I didn't post the urls here, I wasn't sure if that would be frowned upon or against any rules.


I'll remove them if an issue:
https://nordvpn.com
https://protonvpn.com
https://mullvad.net
https://expressvpn.com
https://surfshark.com
https://cyberghostvpn.com
https://privateinternetaccess.com
https://adguard-vpn.com/
https://airvpn.org/

They're all still blocked for me today too

**Comment by drm20000** [comment]
I have no issue to access any of the sites you listed. I am in the US and using a miami serverOn May 24, 2026, at 14:55, David Knight ***@***.***> wrote:﻿DaveRune left a comment (NordSecurity/nordvpn-linux#551)
@v-ip-er I just noticed that I couldn't access nordvpn.com but I could other sites. Out of curiosity I tried other VPN sites and they didn't work. I found a list of VPNs and tried them all, all blocked.
I didn't post the urls here, I wasn't sure if that would be frowned upon or against any rules.
I'll remove them if an issue:
https://nordvpn.com
https://protonvpn.com
https://mullvad.net
https://expressvpn.com
https://surfshark.com
https://cyberghostvpn.com
https://privateinternetaccess.com
https://adguard-vpn.com/
https://airvpn.org/
They're all still blocked for me today too

**Comment by v-ip-er** [comment]
So, small update from me.

1) amazon came back within about 12hrs. I did shift to a new UDP node on my router 2 days ago, but everything is currently stable.
_- FYI: @DoYouEvenCpp, @mariusSincovici_ 

2) circumstance for me @DaveRune re all the sites you listed, I can browse to all of them no problem at all.

Cheers,

**Comment by v-ip-er** [comment]
Famous last words chaps.

My Amazon connectivity is broken again as of this afternoon.

Can't browse to amazon.co.uk, all my devices won't reach their network, can't watch anything on Prime Video.
_FYI @DoYouEvenCpp, @mariusSincovici_

**Comment by drm20000** [comment]
I am un usa.  Amazon.com works fine for me.  But when I try amazon.uk it fails.  So i can confirm your problemOn May 26, 2026, at 13:22, v-ip-er ***@***.***> wrote:﻿v-ip-er left a comment (NordSecurity/nordvpn-linux#551)
Famous last words chaps.
My Amazon connectivity is broken again as of this afternoon.
Can't browse to amazon.co.uk, all my devices won't reach their network.
FYI @DoYouEvenCpp, @mariusSincovici

**Comment by mariusSincovici** [comment]
Hi,

thanks for your feedback. We've notified the backend about this and we'll let you know as soon as there are any updates.

Until then could you let us know on what servers are you connected?
Thanks.

**Comment by v-ip-er** [comment]
I have tried only two;

uk2481.nordvpn.com.udp.ovpn
uk2955.nordvpn.com.tcp.ovpn

Thanks

**Comment by mariusSincovici** [comment]
Could you try now if the issues is still reproducing on your side? 
Because they weren't able to reproduce it.

In case it is. does it happen the same with another server?

Thanks

**Comment by drm20000** [comment]
Amazon.uk is not working for me using nord server us9392 open vpn tcp and Server us6763 openvpn.Amazon.com functions as expectedOn May 28, 2026, at 08:49, Marius ***@***.***> wrote:﻿mariusSincovici left a comment (NordSecurity/nordvpn-linux#551)
Could you try now if the issues is still reproducing on your side?
Because they weren't able to reproduce it.
In case it is. does it happen the same with another server?
Thanks

**Comment by v-ip-er** [comment]
I switched to uk3058.nordvpn.com.udp.ovpn this morning @mariusSincovici, things seem stable... for now!

**Comment by drm20000** [comment]
I just checked amazon.uk.  Still not working for us9392 server tcp and us11809 tcpAmazon.com is reachableOn May 28, 2026, at 15:22, v-ip-er ***@***.***> wrote:﻿v-ip-er left a comment (NordSecurity/nordvpn-linux#551)
I switched to uk3058.nordvpn.com.udp.ovpn this morning @mariusSincovici, things seem stable... for now!

**Comment by Dalavich** [comment]
I've been brought to this discussion, as I have the exact same issue. But I've just run the Nord Diagnostics and reset the Network then reset the App, and so far touch wood Google and other websites are resolving correctly. Oddly everything was working in Edge, just the laptop, Firefox etc wasn't.

**Comment by mariusSincovici** [comment]
Hi,

I'm closing this issue because the discussion has expanded to cover multiple topics and platforms.
If you're experiencing issues with the NordVPN Linux application, please open a separate issue for each specific problem. For issues on other platforms, please contact our Customer Support team, as they'll be able to assist you more effectively.

Thank you for your feedback
